File: app/src/main/java/fr/xgouchet/chronorg/ui/presenters/ProjectListPresenter.java
package fr.xgouchet.chronorg.ui.presenters;

import android.support.annotation.NonNull;

import com.deezer.android.counsel.annotations.Trace;

import javax.inject.Inject;

import fr.xgouchet.chronorg.data.models.Project;
import fr.xgouchet.chronorg.data.repositories.ProjectRepository;
import rx.Observable;

/**
 * @author Xavier Gouchet
 */
@Trace
public class ProjectListPresenter extends BaseListPresenter<Project> {

    @NonNull private final ProjectRepository projectRepository;

    @Inject
    public ProjectListPresenter(@NonNull ProjectRepository projectRepository) {
        this.projectRepository = projectRepository;
    }

    @Override protected Observable<Project> getItemsObservable() {
        return projectRepository.getAll();
    }
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/ui/presenters/ProjectListPresenter.kt
package fr.xgouchet.khronorg.ui.presenters

import fr.xgouchet.khronorg.data.models.Project
import fr.xgouchet.khronorg.data.repositories.BaseRepository
import fr.xgouchet.khronorg.ui.navigators.ProjectNavigator
import io.reactivex.Observable

/**
 * @author Xavier F. Gouchet
 */
class ProjectListPresenter(val repository: BaseRepository<Project>, navigator: ProjectNavigator) : BaseListPresenter<Project>(navigator) {


    override fun getItemsObservable(): Observable<Project> {
        return repository.getAll()
    }

}